New package: SimulationBasedInference v0.1.0

JuliaRegistrator opened the pull request:
- Registering package: SimulationBasedInference
- Repository: https://github.com/bgroenks96/SimulationBasedInference.jl
- Created by: @bgroenks96
- Version: v0.1.0
- Commit: 7da58d9bf9e4b7fff4982e4a55bea1ec3611f1d8
- Reviewed by: @bgroenks96
- Reference: https://github.com/bgroenks96/SimulationBasedInference.jl/commit/7da58d9bf9e4b7fff4982e4a55bea1ec3611f1d8#commitcomment-141036787
- Description: A flexible toolkit for simulation based inference in Julia
<!-- bf0c69308befbd3ccf2cc956ac8a46712550b79fc9bfb5e4edf8f833f05f4c18b06eddad8845b45beb9f45c2b8020dd615ed2455506c7ac8cf3f87c609a5ebe58ccd8a3bb33d7ad3348cde2f4e9d29884fdbe3a09a37255333e381692c87556d898c28f9be02ed8c32359d66bfcc13f431e8765492a99fe85e7e44569f30f54e81e3ac73b999fde5184725605eecf93331ef0ce1752ecaaee2d934c383aea0defd9268d120d161024833ca57bc87caf8c8190bb2b1fff1d382e6cf845fad40687de55cd767c7163407e190d8a7dc874a7b2f36946b55669dae22514a1aeb432a615a7648bbdedf9c4f404528b0aaefc1 -->

gdalle commented:
[noblock]
Hi @bgroenks96 and congrats on the new package! Can I ask how it aims to differentiate itself from other similar initiatives, most notably [Turing](https://turinglang.org/stable/) and its MCMC ecosystem`

JuliaRegistrator committed:
New package: SimulationBasedInference v0.1.0

UUID: 78927d98-f421-490e-8789-96b006983a5c
Repo: https://github.com/bgroenks96/SimulationBasedInference.jl.git
Tree: 26b08f40d37fffd6e24a396e01e0c3f88916b372

Registrator tree SHA: 17aec322677d9b81cdd6b9b9236b09a3f1374c6a

bgroenks96 commented:
Hi @gdalle,

Thanks for your interest.

I am a long time user of Turing, and I love it's simplicity and ease-of-use for many statistical modeling use cases.

However, after several years of use, I must say that Turing is woefully inadequate for use-cases where the forward model is computationally expensive. The interface is not designed in a way that makes sense for such models. For example, it lacks the ability to save and recover deterministic outputs of the model; one must call `generated_quantities` which requires re-running the model which is extremely wasteful in cases where doing so is non-trivial. Furthermore, it is sorely lacking in non-gradient based sampling methods, especially when it comes to parallel sampling (i.e. ensemble or particle-based algorithms, not multi-chain). `SimulationBasedInference` aims to fill this gap.

The closest existing package to this one is probably [DataAssim](https://github.com/Alexander-Barth/DataAssim.jl) but this is focused on classical data assimilation methods, i.e. filtering problems applied to model state, rather than simulation-based Bayesian inference more broadly.

I will add a README section soon to highlight these points.
